src/objectExplorer/objectExplorerService.ts

```diff
@@ -302,6 +302,10 @@ export class ObjectExplorerService {
             const connectionDetails = ConnectionCredentials.createConnectionDetails(connectionCredentials);
             const response = await this._connectionManager.client.sendRequest(CreateSessionRequest.type, connectionDetails);
             if (response) {
+                // remove password from node if saved password is false
+                if (!(<IConnectionProfile>connectionCredentials).savePassword) {
+                    connectionCredentials.password = '';
```

Comment:
This shouldn't be necessary - at the top of this function the shouldPromptForPassword should handle the logic correctly to prompt if disconnected/reconnected. Is that not happening?

Reply:
This logic doesn't have to do anything with `shouldPromptForPassword`. This is to respect the `savePassword` option for the nodes, while getting the password from the prompt. With this, we will have the password per session.

Reply:
I don't get why this is necessary though. save password would be more referring to saving it in the connection store in my eyes. Whether the node has the password or not really shouldn't matter. Is there a bug you're fixing here?

---
In reply to: [329843754](https://github.com/microsoft/vscode-mssql/pull/1356#discussion_r329843754) [](ancestors = 329843754)

Reply:
The `savePassword` option will be respected when the connection is made using that profile. The handleConnectionSuccessNotificationHandler saves any password if `savePassword` was set to true. 

However, when since each node has a connectionCredential, we don't want to populate its password field if `savePassword` is true, but rather look up the credential store for a saved password instead.

Reply:
Well - I would expect that when we made the connection we'd do the lookup for the password then and if it wasn't saved to the credentialStore then it'd prompt the user at that point. So it really shouldn't matter.

I guess in the end this is fine but again - why are you making this change? I don't really see a problem with it or anything but I'm trying to figure out if you're fixing a bug here. 

And then in addition - why would it be necessary to also clear it in the disconnect logic? Is there ever a scenario where the password would be still there if you're stripping it out when you connect here?

Reply:
The line you quoted is just solving the issue you asked about. We need to keep node.connectionCredentials different from connectionCredentials, because we don't want to save that password on the node but to fetch it from credential store.

Reply:
I removed this as it was redundant, you were right. Didn't really need it.